internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go

```diff
@@ -548,6 +558,40 @@ func (h *Handler) removePublicShare(w http.ResponseWriter, r *http.Request, shar
 	response.WriteOCSSuccess(w, r, nil)
 }
 
+// enforcePassword validate Password enforce based on configuration
+// read_only:           1
+// read_write:          3 or 5
+// read_write_delete:   15
+// upload_only:         4
+func (h *Handler) enforcePassword(pk *int) bool {
+	var enforce bool
+	if pk == nil {
+		return enforce
+	}
+	p, err := conversions.NewPermissions(decreasePermissionsIfNecessary(*pk))
+	if err != nil {
+		return enforce
+	}
+	if h.publicPasswordEnforced.EnforcedForReadOnly &&
+		p == conversions.PermissionRead {
+		enforce = true
```

Comment:
Looks correct now. But I have one little nitpick:
We could `return true` here because the `enforce` value will never be changed back to `false` again. If we do this for all `if` cases we can completely get rid of the `enforce` var